tools/gulptasks/dist-productsjs.js

```diff
@@ -26,45 +24,53 @@ const TARGET_FILE = Path.join('build', 'dist', 'products.js');
  * @return {Promise<void>}
  *         Promise to keep
  */
-function distProductsJS() {
+async function distProductsJS() {
 
-    const Fs = require('fs');
+    const fs = require('fs');
     const LogLib = require('./lib/log');
 
-    return new Promise(resolve => {
+    LogLib.message('Generating', TARGET_FILE + '...');
 
-        LogLib.message('Generating', TARGET_FILE + '...');
+    const buildProperties = require('../../build-properties.json');
+    const packageJson = require('../../package.json');
 
-        const buildProperties = require('../../build-properties.json');
-        const packageJson = require('../../package.json');
+    const date = (
+        buildProperties.date ||
+        ''
+    );
 
-        const date = (
-            buildProperties.date ||
-            ''
-        );
+    const nr = (
+        buildProperties.version ||
+        packageJson.version ||
+        ''
+    ).split('-')[0];
 
-        const nr = (
-            buildProperties.version ||
-            packageJson.version ||
-            ''
-        );
+    const dashboardsProduct = {
+        date,
+        nr: '1.0.0'
+    };
 
-        Fs.writeFileSync(
-            TARGET_FILE,
-            (
-                'var products = ' + JSON.stringify({
-                    Highcharts: { date, nr },
-                    'Highcharts Stock': { date, nr },
-                    'Highcharts Maps': { date, nr },
-                    'Highcharts Gantt': { date, nr }
-                }, undefined, '    ') + '\n'
-            )
+    if (fs.existsSync('../dashboards-dist')) {
+        dashboardsProduct.nr = (
+            require('../../../dashboards-dist/package.json').version ||
+            dashboardsProduct.nr
         );
-
-        LogLib.success('Created', TARGET_FILE);
-
-        resolve();
-    });
+    }
+
+    await fs.promises.writeFile(
+        TARGET_FILE,
+        (
+            'var products = ' + JSON.stringify({
+                Highcharts: { date, nr },
+                'Highcharts Stock': { date, nr },
+                'Highcharts Maps': { date, nr },
+                'Highcharts Gantt': { date, nr },
+                'Highcharts Dashboards': dashboardsProduct
+            }, void 0, '    ') + '\n'
+        )
+    );
```

Comment:
I am not sure what it's used for, but won't this overwrite the date for the Highcharts(& family) release when doing a Dashboards release?

Reply:
Oh wait, there's two tasks

Reply:
It is a bit messy. Maybe it is better to download the latest `products.js` from code.highcharts.com and modify it?

Reply:
What do you think about fetching the latest `products.js`?

Reply:
Not sure. I guess it could help avoid problems with the local dist-repo not being updated?

Reply:
Yes, that was the idea. The issue though can be, that the cache on the server is not immediately updated. So releases of several products would need a time buffer in-between. Or the cache purge needs to be automated.

Reply:
Are we able to clear cache after release or only Sam can do it?

Reply:
I know at least three more people that can do it.